src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,44 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (obj.getClass().equals(Rectangle.class)) {
+            Rectangle current = (Rectangle) obj;
+            return compareField(this.width, current.width)
+                    && compareField(this.length, current.length)
+                    && compareField(this.color, current.color);
+        }
+        return false;
+    }
+
+    private boolean compareField(Integer first, Integer second) {
+        if (first == null && second != null || first != null && second == null) {
+            return false;
+        }
+        return first == null || first.equals(second);
+    }
+
+    private boolean compareField(String first, String second) {
```

Comment:
code duplication with the previous method. I guess we can avoid writing `compareField` methods. We can write all logic in `equals` methods